src/init.cpp

```diff
@@ -1768,10 +1774,9 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     fs::path est_path = GetDataDir() / FEE_ESTIMATES_FILENAME;
     CAutoFile est_filein(fsbridge::fopen(est_path, "rb"), SER_DISK, CLIENT_VERSION);
     // Allowed to fail as this file IS missing on first startup.
-    if (!est_filein.IsNull())
-        ::feeEstimator.Read(est_filein);
-    fFeeEstimatesInitialized = true;
```

Comment:
I think this flag might protect us from overwriting the estimates.dat file in case we abort start-up for some reason. Rather than removing it entirely, I think it should be moved to CBlockPolicyEstimator, and then checked on destruction.

If you really wanted to encaptulate this better, you could move this file read logic to the feeEstimator constructor, and the flush logic to a public flush method in CBlockPolicyEstimator, and hide all the construction/destruction logic from init.cpp.

Reply:
Good catch. Went for the encapsulation way, will push a commit.

Reply:
Moving this code to the destructor may be a behaviour change, since there will have been changes to the mempool in the intervening period. I suggest adding a public flush method and calling it here to avoid the behaviour change.

Reply:
Yeah, i initially swapped the two calls to `reset()`, but as you prefer.